clusterdock/topologies/cdh/actions.py

```diff
@@ -215,3 +281,110 @@ def remove_files(cluster, files, nodes):
+    _change_config_files(kdc_node, network)
+    _setup_kdc_server(cluster, [kdc_node], network.upper(), kerberos_principals)
+
+def _change_config_files(kdc_node, network):
+    logger.info('Changing config. files on kdc node...')
+
+    client = Client()
+    config_files = {}
+
+    for kdc_conf_file in KDC_CONFIG_FILES:
+	# docker.Client.get_archive returns a tuple containing the raw tar data stream and a
+        # dict of stat information on the specified path. We only care about the former, so
+	# we discard the latter.
+        tarstream = io.BytesIO(client.get_archive(container=kdc_node.container_id,
+                                                  path=kdc_conf_file)[0].read())
+        with tarfile.open(fileobj=tarstream) as tarfile_:
+            for tarinfo in tarfile_.getmembers():
+                # tarfile.extractfile returns a file object, which, when read, returns a
+                # bytes object.
+                config_files[kdc_conf_file] = tarfile_.extractfile(tarinfo).read().decode()
+
+    # Update configurations
+    krb5_conf_data = config_files[KDC_KRB5_CONF_FILENAME]
+    krb5_conf_data = krb5_conf_data.replace('EXAMPLE.COM', network.upper())
+    krb5_conf_data = krb5_conf_data.replace('kerberos.example.com', '{}.{}'.format(KDC_HOST_NAME, network))
+    krb5_conf_data = krb5_conf_data.replace('example.com', network)
+    config_files[KDC_KRB5_CONF_FILENAME] = krb5_conf_data
+
+    kdc_conf_data = config_files[KDC_CONF_FILENAME]
+    kdc_conf_data = kdc_conf_data.replace('EXAMPLE.COM', network.upper())
+    conf_list = kdc_conf_data.split('\n')
+    index = conf_list.index('[kdcdefaults]')
+    conf_list.insert(index + 1, ' max_renewablelife = 7d')
+    conf_list.insert(index + 1, ' max_life = 1d')
+    kdc_conf_data = '\n'.join(conf_list)
+    config_files[KDC_CONF_FILENAME] = kdc_conf_data
+
+    acl_data = config_files[KDC_ACL_FILENAME]
+    acl_data = acl_data.replace('EXAMPLE.COM', network.upper())
+    config_files[KDC_ACL_FILENAME] = acl_data
+
+    # Serialize them back to the kdc Docker container
+    for kdc_conf_file in KDC_CONFIG_FILES:
+        o_tarstream = io.BytesIO()
+        with tarfile.open(fileobj=o_tarstream, mode='w') as tarfile_:
+            encoded_file_contents = config_files[kdc_conf_file].encode()
+            tarinfo = tarfile.TarInfo(basename(kdc_conf_file))
+            tarinfo.size = len(encoded_file_contents)
+            tarfile_.addfile(tarinfo, io.BytesIO(encoded_file_contents))
+        o_tarstream.seek(0)
+        client.put_archive(container=kdc_node.container_id, path=dirname(kdc_conf_file), data=o_tarstream)
+
+def _setup_kdc_server(cluster, nodes, network, kerberos_principals):
+    logger.info('Setting up kdc server ...')
+    principal_cmds = ""
+    if kerberos_principals:
+        princ_list = kerberos_principals.split(',')
+        create_principals = " ".join(["kadmin.local -q \"addprinc -randkey %s@%s\"; " % (item, network)
```

Comment:
If you're creating a string that will contain a command to create principals, I'd call it `create_principals_command` to avoid confusion on what the variable refers to. (i.e. if I just see `create_principals` and don't see how it's assigned, I'd assume it was a function that will actually create the principals instead of being a string that holds the command).